modules/data/src/main/kotlin/org/gjt/jclasslib/structures/ClassFile.kt

```diff
@@ -452,8 +453,8 @@ class ClassFile : Structure(), AttributeContainer {
     }
 
     private fun checkMajorVersion(majorVersion: Int) {
-        if (majorVersion < 45 || majorVersion > 53) {
-            warning("major version should be between 45 and 53 for JDK <= 1.9, was $majorVersion")
+        if (majorVersion < 45 || majorVersion > 54) {
+            warning("major version should be between 45 and 54 for JDK <= 1.10, was $majorVersion")
```

Comment:
Can you add 55 as well (I have added the support for Java 11 already) and change the message to "... JDK <= 11" (not 1.11)